src/content/docs/en/guides/authentication.mdx

```diff
@@ -0,0 +1,192 @@
+- [Example Astro + Lucia username and password project](https://github.com/lucia-auth/examples/tree/main/astro/username-and-password)
+
+## Auth.js
+
+Auth.js is a framework agnostic solution for authentication. A community framework adapter [`auth-astro`](https://www.npmjs.com/package/auth-astro) is available for Astro.
+
+### Installation
+
+Using `astro add` you can add the integration automatically.
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npx astro add auth-astro
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm astro add auth-astro
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn astro add auth-astro
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+#### Manual installation
+
+First, install the required packages:
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npm install auth-astro
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm add auth-astro @auth/core
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn add auth-astro
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+Then, apply the integration to your `astro.config.*` file using the `integrations` property:
+
+```ts title="astro.config.mjs" ins={2,6}
+import { defineConfig } from 'astro/config';
+import auth from 'auth-astro';
+
+export default defineConfig({
+  // ...
+  integrations: [auth()],
+});
+```
+
+### Configuration
+
+Create an `auth.config.mjs` file in your project's root directory. Add any auth [providers](https://authjs.dev/getting-started/providers) or methods you wish to support, along with any environment variables they require.
+
+```ts title="auth.config.mjs"
+import GitHub from '@auth/core/providers/github'
+import { defineConfig } from 'auth-astro'
+
+export default defineConfig({
+	providers: [
+		GitHub({
+			clientId: import.meta.env.GITHUB_CLIENT_ID,
+			clientSecret: import.meta.env.GITHUB_CLIENT_SECRET,
+		}),
+	],
+})
+```
+
+Add two additional environment variables. `AUTH_SECRET` should be a private string with a minimum of 32 characters.
+
+```sh
+AUTH_TRUST_HOST=true
+AUTH_SECRET=<my-auth-secret>
+```
+
+### Usage
+
+You can add sign-in and sign-out buttons using the `auth-astro/client` module in a script tag or client-side framework component.
+
+```astro title="src/pages/index.ts"
+---
+import Layout from 'src/layouts/Base.astro'
+---
+<Layout>
+  <button id="login">Login</button>
+  <button id="logout">Logout</button>
+
+  <script>
+    const { signIn, signOut } = await import("auth-astro/client")
+    document.querySelector("#login").onclick = () => signIn("github")
+    document.querySelector("#logout").onclick = () => signOut()
+  </script>
+</Layout>
+```
+
+You can fetch the user's session using the `getSession` method.
+
+```astro title="src/pages/index.ts"
+---
+import Layout from 'src/layouts/Base.astro'
+import { getSession } from 'auth-astro/server';
```

Comment:
```suggestion
import { getSession } from 'auth-astro/server'
```

Reply:
I believe this should be the other way around. Semicolons for days!

Reply:
I also prefer semicolons but I suggested without because the rest of the code examples don't use them (for consistency).

Reply:
> I also prefer semicolons but I suggested without because the rest of the code examples don't use them (for consistency).

That makes sense! The documentation tends to (and should) include semicolons in code examples, and part of my fight is finding instances where this is not true! :)